New wipBatch shouldn't store txs

In `v0.0.3-RC17`:

```log
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11d4032]

goroutine 145 [running]:
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).handleSuccessfulTxProcessResp(0xc00055cf00, {0x1918478, 0xc0000c4000}, 0x0, 0xc017efaa90)
	/src/sequencer/finalizer.go:389 +0x152
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).processTransaction(0xc00055cf00, {0x1918478, 0xc0000c4000}, 0x0)
	/src/sequencer/finalizer.go:363 +0x7a5
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).newWIPBatch(0xc00055cf00, {0x1918478?, 0xc0000c4000})
	/src/sequencer/finalizer.go:260 +0xf5
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).finalizeBatch(0xc00055cf00, {0x1918478, 0xc0000c4000})
	/src/sequencer/finalizer.go:244 +0xb0
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).finalizeBatches(0xc00055cf00, {0x1918478, 0xc0000c4000})
	/src/sequencer/finalizer.go:218 +0x33c
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).Start(0xc00055cf00, {0x1918478?, 0xc0000c4000}, 0x56f3aa?, 0x1?)
	/src/sequencer/finalizer.go:131 +0x209
created by github.com/0xPolygonHermez/zkevm-node/sequencer.(*Sequencer).Start
	/src/sequencer/sequencer.go:157 +0xd0c
```

The panic itself is already fixed in develop, but not the underlying problem:

- `newWIPBatch` is called, current batch has no txs, so it's re-executed to ensure GER update is reflected in the new state root.
- `processTransaction` returns successful execution, however there are no txs being processed
- `handleSuccessfulTxProcessResp` is called, but the tx that is passed is `nil`! This function shouldn't be called in this context

So here is were we have the crazy situation: our protocol can alter the state without executing transactions 🤯 : An empty batch with a new GER creates a state update. We need to:

1. Handle this special case by updating the state root of the batch that we're closing, without trying to store transactions and L2 blocks
2. Review that we correctly get the state root for next batches, RPC calls, reboots, ... in this situations

--- a/sequencer/finalizer.go
+++ b/sequencer/finalizer.go
@@ -254,16 +254,6 @@ func (f *finalizer) newWIPBatch(ctx context.Context) (*WipBatch, error) {
 	defer f.sharedResourcesMux.Unlock()
 
 	var err error
-	// Passing the batch without txs to the executor in order to update the State
-	if f.batch.countOfTxs == 0 {
-		// backup current sequence
-		err = f.processTransaction(ctx, nil)
-		for err != nil {
-			log.Errorf("failed to process tx, err: %v", err)
-			err = f.processTransaction(ctx, nil)
-		}
-	}
-
 	if f.batch.stateRoot.String() == "" || f.batch.localExitRoot.String() == "" {
 		return nil, errors.New("state root and local exit root must have value to close batch")
 	}
@@ -360,22 +350,29 @@ func (f *finalizer) processTransaction(ctx context.Context, tx *TxTracker) error
 		return err
 	}
 
-	err = f.handleSuccessfulTxProcessResp(ctx, tx, result)
-	if err != nil {
-		return err
+	oldStateRoot := f.batch.stateRoot
+	if len(result.Responses) > 0 && tx != nil {
+		err = f.handleTxProcessResp(ctx, tx, result, oldStateRoot)
+		if err != nil {
+			return err
+		}
 	}
+
+	// Update in-memory batch and processRequest
+	f.processRequest.OldStateRoot = result.NewStateRoot
+	f.batch.stateRoot = result.NewStateRoot
+	f.batch.localExitRoot = result.NewLocalExitRoot
+	log.Infof("processTransaction: data loaded in memory. batch.batchNumber: %d, batchNumber: %d, result.NewStateRoot: %s, result.NewLocalExitRoot: %s, oldStateRoot: %s", f.batch.batchNumber, f.processRequest.BatchNumber, result.NewStateRoot.String(), result.NewLocalExitRoot.String(), oldStateRoot.String())
 
 	return nil
 }
 
-// handleSuccessfulTxProcessResp handles the response of a successful transaction processing.
-func (f *finalizer) handleSuccessfulTxProcessResp(ctx context.Context, tx *TxTracker, result *state.ProcessBatchResponse) error {
-	if len(result.Responses) > 0 {
-		// Handle Transaction Error
-		if result.Responses[0].RomError != nil && !errors.Is(result.Responses[0].RomError, runtime.ErrExecutionReverted) {
-			f.handleTransactionError(ctx, result, tx)
-			return result.Responses[0].RomError
-		}
+// handleTxProcessResp handles the response of a successful transaction processing.
+func (f *finalizer) handleTxProcessResp(ctx context.Context, tx *TxTracker, result *state.ProcessBatchResponse, oldStateRoot common.Hash) error {
+	// Handle Transaction Error
+	if result.Responses[0].RomError != nil && !errors.Is(result.Responses[0].RomError, runtime.ErrExecutionReverted) {
+		f.handleTransactionError(ctx, result, tx)
+		return result.Responses[0].RomError
 	}
 
 	// Check remaining resources
@@ -384,25 +381,14 @@ func (f *finalizer) handleSuccessfulTxProcessResp(ctx context.Context, tx *TxTra
 		return err
 	}
 
-	previousL2BlockStateRoot := f.batch.stateRoot
 	// Store the processed transaction, add it to the batch and update status in the pool atomically
-	if tx != nil {
-		log.Infof("handleSuccessfulTxProcessResp: storing processed tx: %s", tx.Hash.String())
-	}
-	f.storeProcessedTx(ctx, previousL2BlockStateRoot, tx, result)
-	f.processRequest.OldStateRoot = result.NewStateRoot
-	f.batch.stateRoot = result.NewStateRoot
-	f.batch.localExitRoot = result.NewLocalExitRoot
-	log.Infof("handleSuccessfulTxProcessResp: data loaded in memory. batch.batchNumber: %d, batchNumber: %d, result.NewStateRoot: %s, result.NewLocalExitRoot: %s, previousL2BlockStateRoot: %s", f.batch.batchNumber, f.processRequest.BatchNumber, result.NewStateRoot.String(), result.NewLocalExitRoot.String(), previousL2BlockStateRoot.String())
+	f.storeProcessedTx(ctx, oldStateRoot, tx, result)
 
 	return nil
 }
 
 func (f *finalizer) storeProcessedTx(ctx context.Context, previousL2BlockStateRoot common.Hash, tx *TxTracker, result *state.ProcessBatchResponse) {
-	if tx == nil || len(result.Responses) == 0 {
-		return
-	}
-
+	log.Infof("storeProcessedTx: storing processed tx: %s", tx.Hash.String())
 	f.txsStore.Wg.Wait()
 	txResponse := result.Responses[0]
 	f.txsStore.Wg.Add(1)
@@ -654,6 +640,7 @@ func (f *finalizer) openWIPBatch(ctx context.Context, batchNum uint64, ger, stat
 func (f *finalizer) closeBatch(ctx context.Context) error {
 	// We need to process the batch to update the state root before closing the batch
 	if f.batch.initialStateRoot == f.batch.stateRoot {
+		log.Info("reprocessing batch because the state root has not changed...")
 		err := f.processTransaction(ctx, nil)
 		if err != nil {
 			return err
@@ -784,10 +771,6 @@ func (f *finalizer) isDeadlineEncountered() bool {
 
 // checkRemainingResources checks if the transaction uses less resources than the remaining ones in the batch.
 func (f *finalizer) checkRemainingResources(result *state.ProcessBatchResponse, tx *TxTracker) error {
-	if len(result.Responses) == 0 {
-		return nil
-	}
-
 	usedResources := batchResources{
 		zKCounters: result.UsedZkCounters,
 		bytes:      uint64(len(tx.RawTx)),

